### PR TITLE
Feature: Devices API

### PR DESCRIPTION
**Pull request by sgammon**
This changeset adds support for the _Devices API_ to the Swift API client library.

Two methods are provided by this service:
- `ping`: warmup/test connection with the Devices API.
- `activate`: Query for a device's assignment status.

### CODE REVIEW

File: Sources/Client/Bindings.swift [hunk 1]
```diff
@@ -10,6 +10,27 @@ import Schema
...
+/**
+ * Method: `ping`.
+ */
+public typealias DevicesPing = Bloombox_Schema_Services_Devices_V1beta1_Ping
+
+/**
+ * Method: `activate`.
+ */
+public typealias DeviceActivation = Bloombox_Schema_Services_Devices_V1beta1_Activation
+public typealias DeviceActivateCall = Bloombox_Schema_Services_Devices_V1Beta1_DevicesActivateCall
```
Comment (sgammon) [inline Sources/Client/Bindings.swift]: added bindings for `Schema` and `Client` modules

File: Sources/Client/DeviceClient.swift [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+/**
+ * Provides functionality for the Device API, which supports operations related to activating,
+ * coordinating, managing, and observing managed or second-party-colocated devices.
+ */
+public final class DevicesClient: RemoteService {
+  /**
+   * Name of the Device API, which is "devices".
+   */
+  let name = "devices"
```
Comment (sgammon) [inline Sources/Client/DeviceClient.swift]: added `DevicesClient`

File: Sources/Client/DevicesClient.swift [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+/**
+ * Provides functionality for the Device API, which supports operations related to activating,
+ * coordinating, managing, and observing managed or second-party-colocated devices.
+ */
+public final class DevicesClient: RemoteService {
+  /**
+   * Name of the Device API, which is "devices".
+   */
+  let name = "devices"
```
Comment (sgammon) [inline Sources/Client/DevicesClient.swift]: new `DevicesClient`

File: Sources/Client/DevicesClient.swift [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+  // MARK: - Public API -
+
+  // MARK: Device Activation
+  /**
+   * Activate a device by name.
+   */
+  public func activate(deviceSerial name: DeviceSerial,
+                       withFingerprint fingerprint: DeviceFingerprint,
+                       authorizedBy apiKey: APIKey? = nil) throws -> DeviceActivation.Response {
```
Comment (sgammon) [inline Sources/Client/DevicesClient.swift]: `activate` method, synchronous implementation

File: Sources/Client/DevicesClient.swift [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  /**
+   * Activate a device by name, asynchronously.
+   */
+  public func activate(deviceSerial name: DeviceSerial,
+                       withFingerprint fingerprint: DeviceFingerprint,
+                       authorizedBy apiKey: APIKey? = nil,
+                       _ callback: @escaping DeviceActivateCallback) throws -> DeviceActivateCall {
+    let service = self.service(try resolveContext(apiKey))
```
Comment (sgammon) [inline Sources/Client/DevicesClient.swift]: `activate`, but async

**Comment by tyler-porras** [comment]
Device assignment made easy(er).

**Pull request merged**